client/store/modules/auth/mutations.js

```diff
@@ -1,9 +1,9 @@
-export const login = (state, user) => {
-  state.user = user;
+export const login = (state, data) => {
```

Comment:
This is a bit unreadable, specially because you are assigning to whole state. Consider destructuring payload `{ user, token }` and assign it explicitly. This will also avoid any unwanted assignments.